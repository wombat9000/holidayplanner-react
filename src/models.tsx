export interface AppUser {
    displayName: string;
}