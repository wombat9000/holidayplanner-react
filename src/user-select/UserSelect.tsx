import * as React from 'react';
import {AppUser} from '../models';
import {Button} from '@material-ui/core';
import {UserContext} from '../UserContext';
import {Link} from 'react-router-dom';

const users: string[] = ['Wombat', 'Hedgehog', 'Moose'];

interface Props {
    onUserSelect: (user: AppUser) => void;
}

interface SelectorProps {
    displayName: string;
    handler: () => void;
}

export const UserSelector = (props: SelectorProps) => (
    <Button variant={'contained'} onClick={props.handler}>{props.displayName}</Button>
);

export const UserSelect = (props: Props) => {
    const selectUser = (name: string) => () => props.onUserSelect({displayName: name});
    const selectors = users.map((it, index: number) => <UserSelector key={index}
                                                                     displayName={it}
                                                                     handler={selectUser(it)}/>);

    return (
        <div>
            <UserContext.Consumer>
                {user => user ?
                    (<div>
                        <h1>Welcome {user.displayName}</h1>
                        <Link to={'/the-holiday'}>Go to Holiday</Link>
                    </div>) :
                    (<div>
                        <h1>UserSelect</h1>
                        {selectors}
                    </div>)
                }
            </UserContext.Consumer>
        </div>
    );
};
