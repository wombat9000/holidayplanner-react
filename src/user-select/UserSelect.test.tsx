import {shallow, ShallowWrapper} from 'enzyme';
import {UserSelector} from './UserSelect';
import * as React from 'react';
import {AppUser} from '../models';

beforeEach(() => {
    jest.resetModules();
});

const getSelectorWithContext = (context = {}) => {
    jest.doMock('./../UserContext.tsx', () => {
        return {
            UserContext: {
                Consumer: (props: any) => props.children(context)
            }
        };
    });

    return require('./UserSelect').UserSelect;
};

xdescribe('UserSelect', () => {
    let wrapper: ShallowWrapper;
    const spy = jest.fn();

    const theWombat: AppUser = {
        displayName: 'The Wombat'
    };

    beforeEach(() => {
        const UserSelect = getSelectorWithContext();
        wrapper = shallow(<UserSelect onUserSelect={spy}/>, {context: {user: undefined}});
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should show selectors when no user is set', () => {
        const selector = wrapper.find(UserSelector);
        selector.simulate('click');
        expect(spy).toHaveBeenCalledWith(theWombat);
    });
});