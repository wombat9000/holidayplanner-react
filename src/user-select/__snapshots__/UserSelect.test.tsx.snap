// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelect should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onUserSelect={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Consumer>
        [Function]
      </Consumer>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Consumer>
          [Function]
        </Consumer>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": [Function],
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "user": undefined,
    },
  },
}
`;
